website/www/site/content/en/documentation/sdks/yaml-udf.md

```diff
@@ -207,6 +207,73 @@ criteria. This can be accomplished with a `Filter` transform, e.g.
     keep: "col2 > 0"
 ```
 
+## Splitting
+
+It can also be useful to send different elements to different places
+(similar to what is done with side outputs in other SDKs).
+While this can be done with a set of `Filter` operations, if every
+element has a single destination it can be more natural to use a `Split`
+transform instead which send every element to a unique output.
+For example, this will send all elements where `col1` is equal to `"a"` to the
+output `Split.a`.
+
+```
+- type: Split
+  input: input
+  config:
+    destination: col1
+    outputs: ['a', 'b', 'c']
+
+- type: SomeTransform
+  input: Split.a
+  config:
+    param: ...
+
+- type: AnotherTransform
+  input: Split.b
+  config:
+    param: ...
+```
```

Comment:
I'm not sure this syntax is super intuitive. The `destination` parameter in this case is more like the source column, and the outputs double as comparison values - although this is not the case when a function is defined.

Something more like this for the shorthand
```
- type: Split
  input: input
  config:
    on: col1
    outputs: ['a', 'b', 'c']
```

Seems more intuitive (although I am not fixed on those exact parameter names per se), but it does arguably add complexity by having different parameter names.

---

Also this works for strings, but would it work for say, booleans? or ints?
`input: Split.true`?

Reply:
Yeah, I went back and forth on the naming here. I like `on` a lot better. 

As for non-strings, we have two options: either require a string or convert. The latter could get dicey (both for composites, and even for some primitives, e.g. is it `True` or `true`?) so I think I'll stick with the first for now (which can always be relaxed in the future if we decide semantics there).

Reply:
OK, so I changed this to "on" which I like a lot better but unfortunately yaml 1.1 interprets this as `True`(?!). Any other ideas for naming, or should we hack around that?

Reply:
Only other word I can think of is "using" - "split on" and "split using" seem fairly similar, but I definitely prefer "on".

It looks like you have a "hack" already by adding the quotes which works, but I found a similar issue on stack that overrides the way booleans are parsed which we could add to our `SafeLineLoader`
https://stackoverflow.com/a/36463915
which would keep from altering the yaml itself that would ultimately be surfaced in the display_data.

Reply:
BTW, though this hack (not needed any more) works, one big downside is that it would cause issues for any other tools that might want to parse or otherwise manipulate pipeline definitions (which may or may not be part of Beam).